### PR TITLE
fix: applying pass key to endpoints that were missing it

### PR DESCRIPTION
**Pull request by YazeedLoonat**
This PR addresses #4354 

- [x] Addresses the issue in full

## Description
Applying api pass key to the endpoints that were missing it

## How Can This Be Tested/Reviewed?
Double check you are still able to log in to the partner and public sites and just bop around to make sure happy paths still work

## Author Checklist:

- [x] Added QA notes to the issue with applicable URLs
- [x] Reviewed in a desktop view
- [ ] Reviewed in a mobile view
- [ ] Reviewed considering accessibility
- [ ] Added tests covering the changes
- [ ] Made corresponding changes to the documentation
- [ ] Ran `yarn generate:client` and/or created a migration when required

## Review Process:

- Read and understand the issue
- Ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Either (1) explicitly ask a clarifying question, (2) request changes, or (3) approve the PR, even if there are very small remaining changes, if you don't need to re-review after the updates

### CODE REVIEW

**Comment by mcgarrye** [comment]
What is the reason we aren't adding these to the still open listing endpoints?

**Comment by YazeedLoonat** [comment]
Hey @mcgarrye good question! what is left open are listing get endpoints and listings are open for the public to consume. I felt like putting those behind passkey would be a little unnecessary

**Review by mcgarrye — APPROVED** [review]
Let's Go Mets

File: api/src/controllers/script-runner.controller.ts [hunk 1]
```diff
@@ -17,11 +17,12 @@ import { AdminOrJurisdictionalAdminGuard } from '../guards/admin-or-jurisdiction
 import { DataTransferDTO } from '../dtos/script-runner/data-transfer.dto';
 import { BulkApplicationResendDTO } from '../dtos/script-runner/bulk-application-resend.dto';
 import { AmiChartImportDTO } from '../dtos/script-runner/ami-chart-import.dto';
+import { ApiKeyGuard } from '../guards/api-key.guard';
 
 @Controller('scriptRunner')
 @ApiTags('scriptRunner')
 @UsePipes(new ValidationPipe(defaultValidationPipeOptions))
-@UseGuards(OptionalAuthGuard, AdminOrJurisdictionalAdminGuard)
+@UseGuards(ApiKeyGuard, OptionalAuthGuard, AdminOrJurisdictionalAdminGuard)
```
Comment (ColinBuyck) [inline api/src/controllers/script-runner.controller.ts:25]: When we hit the database url to run the script runner, how does it get the passkey?

**Review by ColinBuyck — APPROVED** [review]
LGTM 🐳 Just one question for my own understanding

**Pull request merged**